unCkeyword does not detect all listed keywords

### Acton Version

0.29.1.20260724.22.17.33

### Reproduction code

In CodeGen.hs we have the `unCkeyword` function where known C keywords are listed. The problem is that the keywords are fed into `Data.Set.fromDistinctAscList` to build the set, and that function has a precondition that the input list is strictly ascending. Because the precondition of the `fromDistinctAscList` function is not checked, and our list of keywords is not strictly ordered (entries with `_` prefix are ASCII sorted in between upper and lower case), the function does not "catch" all keywords. I got an error on the `switch` attribute for example.

I guess using `fromDistinctAscList` instead of `fromList` is a performance optimisation because the ordered list can directly be used to form some sort of tree? Then adding an explicit check for the precondition (strictly ordered) would defeat the purpose of using the function in the first place?

```diff
diff --git a/compiler/lib/src/Acton/CodeGen.hs b/compiler/lib/src/Acton/CodeGen.hs
index e359db19..0f8450e4 100644
--- a/compiler/lib/src/Acton/CodeGen.hs
+++ b/compiler/lib/src/Acton/CodeGen.hs
@@ -970,6 +970,20 @@ unCkeyword str
   | str `Data.Set.member` rws       = preEscape str
   | otherwise                       = str
   where rws                         = Data.Set.fromDistinctAscList [
+                                        "_Alignas",
+                                        "_Alignof",
+                                        "_Atomic",
+                                        "_BitInt",
+                                        "_Bool",
+                                        "_Complex",
+                                        "_Decimal128",
+                                        "_Decimal32",
+                                        "_Decimal64",
+                                        "_Generic",
+                                        "_Imaginary",
+                                        "_Noreturn",
+                                        "_Static_assert",
+                                        "_Thread_local",
                                         "alignas",
                                         "alignof",
                                         "auto",
@@ -1009,26 +1023,12 @@ unCkeyword str
                                         "true",
                                         "typedef",
                                         "typeof",
-                                        "typeof_unequal",
+                                        "typeof_unqual",
                                         "union",
                                         "unsigned",
                                         "void",
                                         "volatile",
-                                        "while",
-                                        "_Alignas",
-                                        "_Alignof",
-                                        "_Atomic",
-                                        "_BitInt",
-                                        "_Bool",
-                                        "_Complex",
-                                        "_Decimal128",
-                                        "_Decimal32",
-                                        "_Decimal64",
-                                        "_Generic",
-                                        "_Imaginary",
-                                        "_Noreturn",
-                                        "_Static_assert",
-                                        "_Thread_local"
+                                        "while"
                                       ]
```

### Error output

```
+- install ActonProject
   +- compile lib ActonProject Debug x86_64-linux-gnu.2.27 9 errors
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/juniper_ex.h:97620:92: error: expected member name or ';' after declaration specifiers
    juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switch switch;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/comptest.c:3:10: note: in file included from /__w/comptest/comptest/juniper_ex/out/types/juniper_ex/comptest.c:3:
#include "out/types/juniper_ex/comptest.h"
         ^
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/comptest.h:5:10: note: in file included from /__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/comptest.h:5:
#include "out/types/juniper_ex/juniper_ex.h"
         ^
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/juniper_ex.h:97620:91: error: expected ';' at end of declaration list
    juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switch switch;
                                                                                          ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/comptest.c:3:10: note: in file included from /__w/comptest/comptest/juniper_ex/out/types/juniper_ex/comptest.c:3:
#include "out/types/juniper_ex/comptest.h"
         ^
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/comptest.h:5:10: note: in file included from /__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/comptest.h:5:
#include "out/types/juniper_ex/juniper_ex.h"
         ^
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/juniper_ex.h:97620:92: error: expected member name or ';' after declaration specifiers
    juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switch switch;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:3:10: note: in file included from /__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:3:
#include "out/types/juniper_ex/juniper_ex.h"
         ^
/__w/comptest/comptest/juniper_ex/./out/types/juniper_ex/juniper_ex.h:97620:91: error: expected ';' at end of declaration list
    juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switch switch;
                                                                                          ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:3:10: note: in file included from /__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:3:
#include "out/types/juniper_ex/juniper_ex.h"
         ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:162811:32: error: expected identifier
        return (B_value)(self->switch);
                               ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:162826:12: error: expected identifier or '('
    B_list switch = (($ISNONE0(N_default_switch)) ? B_mk_list(0) : N_default_switch);
           ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:162831:95: error: expected identifier
    ((juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow)(self))->switch = ((juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switch)juniper_exQ_juniper_exQ_junos_ex_conf_root__configuration__protocols__openflow__switchG_new(switch));
                                                                                              ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:162847:27: error: expected identifier
    $step_serialize(self->switch, state);
                          ^
/__w/comptest/comptest/juniper_ex/out/types/juniper_ex/juniper_ex.c:162863:11: error: expected identifier
    self->switch = $step_deserialize(state);
          ^
error: 9 compilation errors

```

--- a/compiler/lib/src/Acton/CodeGen.hs
+++ b/compiler/lib/src/Acton/CodeGen.hs
@@ -969,7 +969,7 @@ mkCident (c:s)
 unCkeyword str
   | str `Data.Set.member` rws       = preEscape str
   | otherwise                       = str
-  where rws                         = Data.Set.fromDistinctAscList [
+  where rws                         = Data.Set.fromList [
                                         "alignas",
                                         "alignof",
                                         "auto",
@@ -1009,7 +1009,7 @@ unCkeyword str
                                         "true",
                                         "typedef",
                                         "typeof",
-                                        "typeof_unequal",
+                                        "typeof_unqual",
                                         "union",
                                         "unsigned",
                                         "void",

--- a/compiler/lib/test/ActonSpec.hs
+++ b/compiler/lib/test/ActonSpec.hs
@@ -2098,6 +2098,9 @@ main = do
       testBoxing env0 ["deact"]
 
     describe "Pass 9: CodeGen" $ do
+      it "escapes C keywords" $
+        map Acton.CodeGen.unCkeyword ["switch", "typeof_unqual", "_Alignas"] `shouldBe`
+          ["A_switch", "A_typeof_unqual", "A__Alignas"]
       testCodeGen env0 ["ints"]
       testCodeGen env0 ["deact"]
       testCodeGen env0 ["lines"]

